src/org/rust/cargo/project/CargoOpenProjectDescriptor.kt

```diff
@@ -0,0 +1,16 @@
+package org.rust.cargo.project
+
+import com.intellij.ide.actions.OpenProjectFileChooserDescriptor
+import com.intellij.openapi.vfs.VirtualFile
+import org.rust.cargo.Cargo
+
+class CargoOpenProjectDescriptor : OpenProjectFileChooserDescriptor(true) {
+
+    override fun isFileVisible(file: VirtualFile, showHiddenFiles: Boolean): Boolean {
+        return super.isFileVisible(file, showHiddenFiles) && (file.isDirectory || Cargo.BUILD_FILE == file.name)
```

Comment:
hmm strange that it didn't work... but definitely a good thing that Kotlin does it that way :smiley: